pkg/apiserver/handlers.go

```diff
@@ -354,6 +355,18 @@ func NewRequestAttributeGetter(requestContextMapper api.RequestContextMapper, re
 	return &requestAttributeGetter{requestContextMapper, &APIRequestInfoResolver{util.NewStringSet(apiRoots...), restMapper}}
 }
 
+func (r *requestAttributeGetter) hasApiPrefix(req *http.Request) bool {
+	has := false
+	// Slice the first / of the path off (in apiRoots they're stored w/o leading /)
+	parts := strings.Split(req.URL.Path, "/")[1:]
+	// Paths with only one fragment won't have a prefix.
+	for i := 1; i < len(parts) && !has; i++ {
+		prefix := strings.Join(parts[:i], "/")
+		has = r.apiRequestInfoResolver.APIPrefixes.Has(prefix)
```

Comment:
If this is true, just return here.  It'll make the method easier to read, since you can `return false` after the loop.  (one less variable on the mental stack.)